apps/browser/qml/pages/components/AddHomePageDialog.qml

```diff
@@ -0,0 +1,63 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Policy 1.0
+import org.nemomobile.configuration 1.0
+
+Dialog {
+    id: root
+    property alias text: textField.text
+    property alias placeholderText: textField.placeholderText
+    property alias description: textField.description
+
+    function containsSpaces(text) {
+       return /^\s+|\s+$|\s*(\s)/.test(text) && text.length > 0
+    }
+
+    canAccept: (textField.text !== "" && !textField.errorHighlight)
+
+    DialogHeader {
+        id: header
+        //: Accept button text for adding a home page adress
+        //% "OK"
+        acceptText: qsTrId("sailfish_browser-he-ok")
+    }
+
+    onAccepted: {
+        homePageConfig.value = textField.text || "about:blank"
+    }
+
+    TextField {
+        id: textField
+
+        width: parent.width
```

Comment:
I believe `TextField` defaults to the parent width, so this can be removed.

Reply:
@llewelld  Done